episodes/import-and-visualise-osm-data.Rmd

```diff
@@ -0,0 +1,226 @@
+
+
+### Projections
+
+First, we are going to select the polygons and reproject them with the Amersfoort/RD New projection, suited for maps centred on the Netherlands. This code for this projection is: 28992.
+
+```{r}
+buildings <- x$osm_polygons %>%
+  st_transform(.,crs=28992)
+```
+
+:::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::: instructor
+
+If this does work for some reason, the `buildings` can be found in the data folder: "episodes/data/dataBrielle.shp".
+
+::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+
+
+
+### Visualisation
+
+Then we create a variable which a threshold at 1900. Every date prior to 1900 will be recoded 1900, so that buildings older than 1900 will be represented with the same shade.
+
+Then we use the `ggplot` function to visualise the buildings by age. The specific function to represent information as a map is `geom_sf()`. The rest works like other graphs and visualisation, with `aes()` for the aesthetics.
+
+```{r}
+buildings$build_date <- as.numeric(
+  if_else(
+    as.numeric(buildings$start_date) < 1900, 
+          1900, 
+          as.numeric(buildings$start_date)
+          )
+  )
+
+ ggplot(data = buildings) +
+   geom_sf(aes(fill = build_date, colour=build_date))  +
+   scale_fill_viridis_c(option = "viridis")+
+   scale_colour_viridis_c(option = "viridis")
+```
+
+So this reveals the historical centre of [city] and the various extensions.
+Anything odd? what? around the centre? Why these limits / isolated points?
+
+
+
+::::::::::::::::::::::::::::::::::::: challenge 
+
+## Challenge: import an interactive basemap layer under the buildings with `Leaflet` (20min)
+
+- Check out the [leaflet package documentation](https://rstudio.github.io/leaflet/)
+- Plot a basemap in Leaflet and try different tiles. [Basemap documentation](https://rstudio.github.io/leaflet/basemaps.html)
+- Transform the buildings into WGS84 projection and add them to the basemap layer with the `addPolygons` function.
+- Have the `fillColor` of these polygons represent the `build_date` variable. [Choropleth documentation](https://rstudio.github.io/leaflet/choropleths.html). Using the example and replace the variable names where needed.
+
+
+:::::::::::::::::::::::: solution 
+
+## One solution
+ 
+```{r}
+buildings2 <- buildings %>%
+  st_transform(.,crs=4326)
+
+leaflet(buildings2) %>%
+# addTiles()
+  addProviderTiles(providers$CartoDB.Positron) %>%
+  addPolygons(color = "#444444", weight = 0.1, smoothFactor = 0.5,
+  opacity = 0.2, fillOpacity = 0.8,
+  fillColor = ~colorQuantile("YlGnBu", -build_date)(-build_date),
+  highlightOptions = highlightOptions(color = "white", weight = 2,
+    bringToFront = TRUE))
+```
+
+:::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: 
+
+::::::::::::::::::::::::::::::::::::: keypoints 
+
+- Use the `Nominatim` and `OverPass` APIs within R
+- Use the `osmdata` and `nominatimlite` packages to retrieve geospatial data
```

Comment:
see my comment about the use of `nominatimlite` above.

Reply:
I used osmdata::getbb() in the first instance of the workshop.
Since then it does not work as it used to, so I switched to nominatim (and also I like to show where it comes from anyway)